projects/dendrites/dendrite_segments.py

```diff
@@ -0,0 +1,167 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+A simple implementation of dendrite segments. This is meant to offer, at the least,
+a template for revision and further modifications.
+"""
+
+import math
+
+import numpy as np
+import torch
+from torch.nn import init
+
+
+class DendriteSegments(torch.nn.Module):
+    """
+    This implements dendrite segments over a set of units. Each unit has a set of
+    segments modeled by a linear transformation from a context vector to output value
+    for each segment.
+
+    :param num_units: number of units i.e. neurons;
+                      each unit will have it's own set of dendrite segments
+    :param num_context: length of the context vector;
+                        the same context will be applied to each segment
+    :param num_segments: number of dendrite segments per unit
+    :param sparsity: sparsity of connections;
+                     this is over each linear transformation from
+                     num_context to num_segments
```

Comment:
should these parameters be defined in the __init__ function's docstring instead?